src/summarizedexperiment/dispatchers/combiners.py

```diff
@@ -0,0 +1,91 @@
+from typing import Union
+from functools import singledispatch
+import pandas as pd
+from biocframe import BiocFrame
+
+
+@singledispatch
+def combine(left, right) -> pd.DataFrame:
+    """Combine various objects.
+
+    Args:
+        x (any): supported object.
+    Raises:
+        NotImplementedError: if type is not supported.
+    Returns:
+        pd.DataFrame: concatenated DataFrame.
```

Comment:
honestly, this is a bit confusing. 

you can set extra parameters in the same `combine` dispatcher for `ignore_names` and to `prefer_left`

Reply:
i agree that would be clearer.

also, slightly off-topic but do you have any suggestions for a 'multiple dispatch' design? for example, in `combine`, i'd like to know the dtype of left and right so that i can handle BiocFrames/pd.DataFrames accordingly. there's a [SO thread](https://stackoverflow.com/questions/66711586/is-there-any-difference-between-multimethod-and-multipledispatch) that provides some ideas, but was wondering if you had any thoughts about it

Reply:
I don't have any opinions either. for now, lets have a single `combine` dispatcher with `ignore_names` and `prefer_left` as params